.github/workflows/clang-format-checker.yml

```diff
@@ -0,0 +1,38 @@
+name: 'PR clang-format checker'
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - synchronize
+      - reopened
+permissions:
+  contents: read
+  pull-requests: write
+  issues: write
+jobs:
+  check-pr-formatting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: install clang-format
+        run: sudo apt install -y clang-format
+      - run: |
```

Comment:
Having no name, this step is named "echo Comparing $GITHUB_BASE_REF vs HEAD without the variable substitution. a more useful name might be slightly more helpful.